connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -348,6 +350,33 @@ public void testWakeupInCommitSyncCausesRetry() throws Exception {
         PowerMock.verifyAll();
     }
 
+    @Test
+    public void testTimestampPropagation() throws Exception {
+        expectInitializeTask();
+        expectPollInitialAssignment();
+
+        expectConsumerPoll(1);
+        expectConvertMessages(1);
+
+        Capture<Collection<SinkRecord>> records = EasyMock.newCapture(CaptureType.ALL);
+
+        sinkTask.put(EasyMock.capture(records));
+
+        PowerMock.replayAll();
+
+        workerTask.initialize(TASK_CONFIG);
+        workerTask.initializeAndStart();
+        workerTask.poll(Long.MAX_VALUE);
+        workerTask.poll(Long.MAX_VALUE);
```

Comment:
Ah I see, thanks. I removed the `expectPollInitialAssignment` since that's not under test here, and indeed now only need the single `poll()` call to capture the record.